webpack/InsightsCloudSync/Components/InsightsSettings/InsightsSettingsActions.js

```diff
@@ -0,0 +1,60 @@
+import API from 'foremanReact/API';
+import { insightsCloudUrl } from '../../InsightsCloudSyncHelpers';
+import {
+  INSIGHTS_SYNC_SETTING_SET,
+  INSIGHTS_SYNC_SETTING_SET_ERROR,
+  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+} from './InsightsSettingsConstants';
+
+export const getInsightsSyncSettings = () => async dispatch => {
+  try {
+    const {
+      data: { insightsSyncEnabled },
+    } = await API.get(insightsCloudUrl('settings'));
+    dispatch({
+      type: INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+      payload: {
+        settings: {
+          insightsSyncEnabled,
+        },
+      },
+    });
+  } catch (error) {
+    dispatch({
+      type: INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+      payload: {
+        settings: {
+          error: error.message,
+        },
+      },
+    });
+  }
+};
+
+export const setInsightsSyncEnabled = currentInsightsSyncEnabled => async dispatch => {
+  try {
+    const {
+      data: { insightsSyncEnabled },
+    } = await API.post(insightsCloudUrl('settings'), {
```

Comment:
should the link update a specific setting, e.g:

```suggestion
    } = await API.post(insightsCloudUrl('insightSync'), {
```

Reply:
The settings component should be responsible for all settings at once - I don't want to create separate calls for each setting.